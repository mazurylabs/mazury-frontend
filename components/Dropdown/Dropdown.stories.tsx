import { ComponentMeta } from '@storybook/react';
import { Button, Input } from 'components';
import React, { useState } from 'react';

import { Dropdown } from './Dropdown';

export default {
  title: 'Components/Dropdown',
  component: Dropdown,
  argTypes: {},
} as ComponentMeta<typeof Dropdown>;

export const Default = () => {
  return (
    <Dropdown label="Label">
      <div>Content</div>
    </Dropdown>
  );
};
