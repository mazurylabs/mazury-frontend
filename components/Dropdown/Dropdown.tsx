import Image from 'next/image';
import { useEffect, useRef, useState } from 'react';
import { FCWithClassName } from 'types';

interface DropdownProps {
  label: string;
  optionsContainerClassName?: string;
}

export const Dropdown: FCWithClassName<DropdownProps> = ({
  className,
  children,
  label,
  optionsContainerClassName,
}) => {
  const optionsContainerRef = useRef<HTMLDivElement>(null);
  const [showOptions, setShowOptions] = useState(false);

  const toggleShowOptions = () => {
    setShowOptions((v) => !v);
  };

  // TODO: handle click outside of options container
  // useEffect(() => {
  //   const handleClickOutside = (event: any) => {
  //     if (
  //       optionsContainerRef.current &&
  //       !optionsContainerRef.current.contains(event.target)
  //     ) {
  //       setShowOptions(false);
  //     }
  //   };
  //   document.addEventListener('click', handleClickOutside, true);
  //   return () => {
  //     document.removeEventListener('click', handleClickOutside, true);
  //   };
  // }, []);

  return (
    <div className="relative">
      <div
        onClick={toggleShowOptions}
        className={`flex w-fit hover:cursor-pointer ${className}`}
        role="dropdown"
      >
        <label
          htmlFor={label}
          className="mr-2 text-sm font-bold text-indigoGray-90"
        >
          {label}
        </label>
        <Image
          src="/icons/angle-down.svg"
          width="16px"
          height="16px"
          alt="Dropdown icon"
        />
      </div>

      {/* Content / options */}
      {showOptions && (
        <div
          ref={optionsContainerRef}
          className={`top-o absolute left-0 rounded-3xl border border-indigoGray-20 bg-white p-6 shadow-xl ${optionsContainerClassName}`}
          data-testid="dropdown-options"
        >
          {children}
        </div>
      )}
    </div>
  );
};
