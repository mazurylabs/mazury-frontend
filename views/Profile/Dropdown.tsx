import * as React from 'react';
import SVG from 'react-inlinesvg';

interface DropdownProps {
  label: string;
  options: string[];
  onSelect: (option: string) => void;
  className?: string;
}

export const Dropdown: React.FC<DropdownProps> = ({
  options,
  onSelect,
  className,
  label,
}) => {
  const [toggleDropdown, setToggleDropdown] = React.useState(false);
  const [selectedOption, setSelectedOption] = React.useState<string>();

  const handleSelect = (option: string) => {
    setSelectedOption(option);
    onSelect(option);
  };

  return (
    <button
      type="button"
      className={`relative flex h-12 items-center justify-between rounded-lg  bg-indigoGray-5 px-4 ${className}`}
      //   onClick={() => setToggleDropdown(!toggleDropdown)}
    >
      <p className="font-sansMid text-sm font-medium text-indigoGray-50">
        {selectedOption || `All ${label.toLowerCase()}`}
      </p>
      <SVG src="/icons/chevron-down.svg" height={24} width={24} />

      {toggleDropdown && (
        <div className="absolute top-0 left-0 z-10 h-[400px] w-full bg-red-200"></div>
      )}
    </button>
  );
};
