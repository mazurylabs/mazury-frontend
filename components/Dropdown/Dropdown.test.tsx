import { fireEvent, render, screen } from '@testing-library/react';
import { Dropdown } from './Dropdown';

const DropdownContent = () => {
  return <div>Content</div>;
};

const DropdownContainer = () => {
  return (
    <Dropdown label="Dropdown label">
      <DropdownContent />
    </Dropdown>
  );
};

test('Dropdown', () => {
  render(<DropdownContainer />);
  const { getByRole, getByTestId } = screen;
  const dropdown = getByRole('dropdown');
  expect(dropdown).toBeInTheDocument();
  fireEvent.click(dropdown);
  const dropdownContent = getByTestId('dropdown-options');
  expect(dropdownContent).toBeInTheDocument();
  expect(dropdown).toMatchSnapshot();
});
