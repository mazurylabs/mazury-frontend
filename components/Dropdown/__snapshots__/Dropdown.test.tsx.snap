// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown 1`] = `
<div
  class="flex w-fit hover:cursor-pointer undefined"
  role="dropdown"
>
  <label
    class="mr-2 text-sm font-bold text-indigoGray-90"
    for="Dropdown label"
  >
    Dropdown label
  </label>
  <span
    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
  >
    <span
      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
      />
    </span>
    <img
      alt="Dropdown icon"
      data-nimg="intrinsic"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
    <noscript />
  </span>
</div>
`;
